dexter/src/main/java/com/karumi/dexter/listener/threaddecorator/MainThread.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter.listener.threaddecorator;
+
+import android.os.Handler;
+import android.os.Looper;
+
+/**
+ * A thread specification to execute passed runnable objects in the main thread
+ */
+class MainThread implements Thread {
+
+  MainThread() {
+  }
+
+  @Override public void execute(Runnable runnable) {
+    if (runningMainThread()) {
+      runnable.run();
+    } else {
+      new Handler(Looper.getMainLooper()).post(runnable);
+    }
+  }
+
+  @Override public void loop() {
+  }
+
+  private static boolean runningMainThread() {
+    return Looper.getMainLooper().getThread() == java.lang.Thread.currentThread();
```

Comment:
If you want, you can remove the usage of ``java.lang.Thread.currentThread()`` changing this if to something like:

```java
   return Looper.myLooper() == Looper.getMainLooper()
```